CollisionOperator/CollisionProcessors/neutronCEimp_class.f90

```diff
@@ -644,27 +657,53 @@ subroutine scatterFromMoving(self, p, collDat, reac)
     class(particle), intent(inout)             :: p
     type(collisionData),intent(inout)          :: collDat
     class(uncorrelatedReactionCE), intent(in)  :: reac
-    integer(shortInt)                          :: MT, nucIdx
+    integer(shortInt)                          :: nucIdx
     real(defReal)                              :: A, kT, mu
     real(defReal),dimension(3)                 :: V_n           ! Neutron velocity (vector)
     real(defReal)                              :: U_n           ! Neutron speed (scalar)
     real(defReal),dimension(3)                 :: dir_pre       ! Pre-collision direction
     real(defReal),dimension(3)                 :: dir_post      ! Post-collicion direction
     real(defReal),dimension(3)                 :: V_t, V_cm     ! Target and CM velocity
     real(defReal)                              :: phi, dummy
+    real(defReal)                              :: maj
+    logical(defBool)                           :: eRange, hasDBRC
+    character(100), parameter :: Here = 'ScatterFromMoving (neutronCEimp_class.f90)'
 
-    ! Read data
-    MT     = collDat % MT
-    nucIdx = collDat % nucIdx
+    ! Read collision data
     A      = collDat % A
     kT     = collDat % kT
+    nucIdx = collDat % nucIdx
 
     ! Get neutron direction and velocity
     dir_pre = p % dirGlobal()
     V_n     = dir_pre * sqrt(p % E)
 
-    ! Sample velocity of target
-    V_t = targetVelocity_constXS(p % E, dir_pre, A, kT, p % pRNG)
+    ! Sample target velocity with constant XS or with DBRC
+    ! Check energy range
+    eRange = ((p % E <= self % DBRCeMax) .and. (self % DBRCeMin <= p % E))
+    ! Check if DBRC is on for this target nuclide
+    hasDBRC = self % nuc % hasDBRC()
+
+    if (eRange .and. hasDBRC) then
+
+      ! Retrieve 0K nuclide index from DBRC nuclide map
+      nucIdx = self % xsData % mapDBRCnuc % get(nucIdx)
+
+      ! Reassign pointer for the 0K nuclide
+      self % nuc => ceNeutronNuclide_CptrCast(self % xsData % getNuclide(nucIdx))
```

Comment:
I am wondering if it wouldn't be petter to have a specific, local pointer (local variable) for the 0K version of the nuclide? 
Basically right now the nuclide which is saved in the processor is being changed if DBRC happens to be applied and I fear that it may cause confusion at some point in the future if somebody tries to access the nuclide to e.g. apply cut-offs or something similar?